### PR TITLE
feature: revert check if total score changes, add check if outcomes has zero value and wasn't externally graded

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/INF-250
## Description:
Currently we change SCORE_TOTAL timestamp only when its value changes, but we need to change it also when we have 0 as outcome value. 
Initial value of outcome if 0 so we can't simply check if value was set or not. Because of that we need to check if outcome wasn't externally graded when we set 0 as value, and pass this information to function which sets SCORE_TOTAL timestamp.

## What's Changed

- Check if outcome has not been externally graded and has zero value.

### CODE REVIEW

File: models/Import/Service/ResultImporter.php [hunk 1]
```diff
@@ -356,4 +366,38 @@ private function getResultStorage(): AbstractRdsResultStorage
...
     }
+
+    /**
+     * Zero don't change SCORE_TOTAL value, but we want to update its timestamp when we have new value
+     * @param ImportResultInput $input
+     * @return bool
+     * @throws ImportResultException
+     * @throws common_exception_Error
+     */
+    private function checkOutcomeWithNewZeroValue(ImportResultInput $input): bool
```
Comment (pnal) [inline models/Import/Service/ResultImporter.php:377]: @Karol-Stelmaczonek I can not understand what this function checks from it's name or even from phpdoc

File: models/Import/Service/ResultImporter.php [hunk 2]
```diff
@@ -71,31 +72,40 @@ function () use ($resultStorage, $input, $testUri, $deliveryExecutionUri, $testS
...
+                    $totalScoreCalculatedByItemOutcomes,
+                    $isOutcomeWithNewZeroValue
                 );
             }
         );
     }
 
     /**
-     * @throws ImportResultException
+     * @param AbstractRdsResultStorage $resultStorage
```
Comment (pnal) [inline models/Import/Service/ResultImporter.php:89]: we do not use phpdoc annotations for parameters where it does give you more information than you can get from parameters list.

File: models/Import/Service/ResultImporter.php [hunk 3]
```diff
@@ -356,4 +366,38 @@ private function getResultStorage(): AbstractRdsResultStorage
...
+                    $resultStorage,
+                    $deliveryExecutionUri,
+                    $itemId,
+                    $callItemId,
+                    $outcomeId
+                );
+
+                /** @var taoResultServer_models_classes_Variable $variable */
+                $variable = $outcomeVariable['variable'];
+                if ($outcomeValue === 0.0 && $variable->getExternallyGraded() === false) {
```
Comment (pnal) [inline models/Import/Service/ResultImporter.php:395]: did not understand how this condition checks that a value is "new" for a variable

**Review by pnal — CHANGES_REQUESTED** [review]

**Review by bartlomiejmarszal — COMMENTED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Comment by Karol-Stelmaczonek** [comment]
Abandoned

**Pull request closed without merging**